### PR TITLE
[autoscaler] Improve argument handling for submit

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Change from `--args="..."` to `-- ...`. This improves user experience.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by richardliaw** [comment]
@hartikainen assigning this to you for now ;) feel free to remove the assignment if you're busy

File: doc/source/tune/guides/plot_tune-distributed.rst [hunk 1]
```diff
@@ -257,12 +256,12 @@ Below are some commonly used commands for submitting experiments. Please see the
...
     # Upload `tune_experiment.py` from your local machine onto the cluster. Then,
     # run `python tune_experiment.py --address=localhost:6379` on the remote machine.
-    $ ray submit CLUSTER.YAML tune_experiment.py --args="--address=localhost:6379"
+    $ ray submit CLUSTER.YAML tune_experiment.py -- --address=localhost:6379
 
     # Start a cluster and run an experiment in a detached tmux session,
     # and shut down the cluster as soon as the experiment completes.
     # In `tune_experiment.py`, set `tune.run(upload_dir="s3://...")` to persist results
-    $ ray submit CLUSTER.YAML --tmux --start --stop tune_experiment.py  --args="--address=localhost:6379"
+    $ ray submit CLUSTER.YAML --tmux --start --stop tune_experiment.py  -- --address=localhost:6379
```
Comment (hartikainen) [inline doc/source/tune/guides/plot_tune-distributed.rst:264]:
```suggestion
    $ ray submit CLUSTER.YAML --tmux --start --stop tune_experiment.py -- --address=localhost:6379
```

File: python/ray/scripts/scripts.py [hunk 2]
```diff
@@ -753,19 +753,28 @@ def rsync_up(cluster_config_file, source, target, cluster_name, all_nodes):
     type=int,
     help="Port to forward. Use this multiple times to forward multiple ports.")
 @click.argument("script", required=True, type=str)
-@click.option("--args", required=False, type=str, help="Script args.")
+@click.option(
+    "--args", required=False, type=str, help="(deprecated) Script args.")
```
Comment (hartikainen) [inline python/ray/scripts/scripts.py:757]:
```suggestion
    "--args", required=False, type=str, help="(deprecated) Use '-- --arg1 --arg2' for script args.")
```

Or something like that.

**Review by hartikainen — APPROVED** [review]
Looks good to me!

**Pull request merged**